packages/react-button/src/components/CompoundButton/CompoundButton.test.tsx

```diff
@@ -8,4 +9,126 @@ describe('CompoundButton', () => {
     Component: CompoundButton as React.FunctionComponent<CompoundButtonProps>,
     displayName: 'CompoundButton',
   });
+
+  describe('when rendered as a button', () => {
+    it('renders correctly', () => {
+      const result = render(<CompoundButton>This is a button</CompoundButton>);
+      const button = result.getByRole('button');
```

Comment:
Something I learned recently: `getByRole` will throw if it doesn't find the element, so a `toBeTruthy` check is technically not necessary. If it might not be there or you prefer to explicitly have a truthy check you can use `queryByRole`.